### ISSUE TITLE
Regression: using a combination of BModals, the page's vertical scrollbar no longer reappears since (at least) version 0.25.2

### ISSUE THREAD

**Issue by tbl0605** [bootstrap-vue-next/bootstrap-vue-next#2272]
### Describe the bug

Hi,

### Reproduction
1. Go to https://stackblitz.com/edit/github-rzxbjv-sm8va4
2. The preview window has a vertical scrollbar
3. Click on the "Open Modal" button
    If you inspect the code source of the preview page you can see that the `body` element has `style="overflow: hidden; padding-right: 17px;"`

![image](https://github.com/user-attachments/assets/c91e2d8d-38a1-4bed-8f6f-c172e2152987)
4. Close the modal using the "OK" button
5. The scrollbar hasn't come back (but it should)
    If you inspect the code source of the preview page you can see that the `body` element still has `style="overflow: hidden; padding-right: 17px;"` !

![image](https://github.com/user-attachments/assets/29122830-0558-4f04-b245-60bab6a84045)

Reverting to version 0.24.23 makes everything work again.

Note that my demo is a bit ugly, it was hard to reproduce this bug easily ^^

Thierry.

### Reproduction

https://stackblitz.com/edit/github-rzxbjv-sm8va4

### Used Package Manager

pnpm

### Validations

- [X] Have tested with the latest version. This is still alpha version and sometime things change rapidly.
- [X] Follow our [Code of Conduct](https://github.com/bootstrap-vue-next/bootstrap-vue-next/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/bootstrap-vue-next/bootstrap-vue-next/blob/main/CONTRIBUTING.md).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by tbl0605** [comment]
Actually first non-working version is 0.25.1

@xvaara,
if it helps, it's probably fix for https://github.com/bootstrap-vue-next/bootstrap-vue-next/pull/2263 that introduced this regression, because it's `useSafeScrollLock` that adds `style="overflow: hidden"`to the body element.

**Comment by xvaara** [comment]
Yes it add's it, but useSafeScrollLock has been there many many versions. That pr, just tried to fix it's other bugs.

In your example it's a really weird way of using.. You are using variable change in v-if, after v-if is bringing the element to the render tree, then waiting a animation frame and using ref.show() to show it. But you're not hiding it' you're removing it from the tree with turning the v-if variable to false. Is there a reason why this is so complex? And not just modelValue and lazy="true" ?

**Comment by tbl0605** [comment]
> Yes it add's it, but useSafeScrollLock has been there many many versions. That pr, just tried to fix it's other bugs.
> 
> In your example it's a really weird way of using.. You are using variable change in v-if, after v-if is bringing the element to the render tree, then waiting a animation frame and using ref.show() to show it. But you're not hiding it' you're removing it from the tree with turning the v-if variable to false. Is there a reason why this is so complex? And not just modelValue and lazy="true" ?

Hello,
First of all: thanks for solving this issue ^^
I originally did this to overcome performance issues in bootstrap-vue and kept it as is in bootstrap-vue-next.
The context:
I have main views with many b-modals inside (either directly or as subcomponents), the b-modals themselves having complex content (like b-tables with a huge list of items). Also, the main views can change (at any time) the list of items that will be shown in the b-modals.
If I don't do `<b-modal v-if="showModal">` the main view will always update the content of that b-modal, even if the b-modal is hidden, which seriously slows down the main view. I simply added a watcher on `showModal`, so when it's value is true it does `$refs.modal.show()`, and when the value of `showModal` is false it simply "destroys" (and thus hides) the b-modal because of the falsy `v-if` condition.
I first tried putting the `v-if="showModal"` inside the b-modal, but that caused some visual glitches when opening the b-modal, so I decided to use `<b-modal v-if="showModal">` which works great.
